### DISCUSSION TITLE
OAuth2 SMTP connection to Microsoft 365 Exchange

### DISCUSSION THREAD

**Discussion by dvomsaal** [jstedfast/MailKit#1793] [General]
I see a lot of online discussion about this combination, including some that state Microsoft is not supporting this yet, but I really need a definitive answer - can MailKit be used for this combination? We have been working on it for a couple days and have not been successful, using scopes of both graph and Mail.Send. There is very little information logged other than 

535 5.7.3 Authentication unsuccessful [xxx.xxx.prod.outlook.com 2024-08-12T21:11:19.941Z 08DCB9CD3DF39524]

after the Client sends the AUTH XOAUTH2 ********

Has anyone ever gotten this to work? To date every response that I can find that says "yes" is actually using IMAP once you read the details ;).

Thanks!

**Comment by jstedfast** [comment]
> can MailKit be used for this combination?

Yes.

> We have been working on it for a couple days and have not been successful, using scopes of both graph and Mail.Send.

The graph scopes are for the HTTP REST (aka Graph) API only.

Follow the directions here: https://github.com/jstedfast/MailKit/blob/master/ExchangeOAuth2.md
